datafusion/expr/src/built_in_function.rs

```diff
@@ -325,6 +325,10 @@ fn function_to_name() -> &'static HashMap<BuiltinScalarFunction, &'static str> {
 impl BuiltinScalarFunction {
     /// an allowlist of functions to take zero arguments, so that they will get special treatment
     /// while executing.
+    #[deprecated(
+        since = "32.0.0",
+        note = "please use TypeSignature::supports_zero_argument instead"
+    )]
     pub fn supports_zero_argument(&self) -> bool {
         matches!(
```

Comment:
Can you also please change this method to call into `self.type_signature().supports_zero_argument` so it is clear they are equivalent and so the implementations don't drift out of sync?

Reply:
This makes sense, updated. Thank you.